librad/src/git/include.rs

```diff
@@ -0,0 +1,239 @@
+// This file is part of radicle-link
+// <https://github.com/radicle-dev/radicle-link>
+//
+// Copyright (C) 2019-2020 The Radicle Team <dev@radicle.xyz>
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 or
+// later as published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use std::{
+    convert::TryFrom,
+    marker::PhantomData,
+    path::{self, PathBuf},
+};
+
+use git2::{Config, ConfigLevel};
+
+use crate::{
+    git::{
+        local::url::LocalUrl,
+        types::{remote::Remote, AsRefspec, FlatRef, Force, Reference},
+    },
+    hash::Hash,
+    meta::user::User,
+    peer::PeerId,
+    uri::RadUrn,
+};
+
+/// An `Include` is a representation of an include file which we want to
+/// generate for working copies.
+///
+/// When generated it will include a list of remotes of the form:
+/// ```text
+/// [remote "<handle>@<peer_id>"]
+///     url = rad://<repo>@<peer_id>.git
+///     fetch = refs/heads/*:refs/remotes/<repo>/*
```

Comment:
This doesn't look right. Wasn't the idea to generate a remote per tracked peer of a project? So I'd expect it to look more like:

```
[remote "<handle>@<peer_id>"]
    url = rad://<local_peer_id_because_libgit>@<urn.id>.git
    fetch = refs/remotes/<peer_id>/refs/heads/*:refs/remotes/<handle>@<peer_id>/*
```

It's possible that the types ensure that, but it's hard to tell from this patch alone.

Reply:
Mmmm you're right. Lemme double-check this!

Reply:
Added a test and found a bug I'd left :sweat_smile: https://github.com/radicle-dev/radicle-link/pull/300/commits/373c80b032dd9bb3a3e19536e6ab17b064b3e7a3